staging/src/k8s.io/apiserver/pkg/storage/etcd3/metrics/metrics.go

```diff
@@ -85,6 +86,46 @@ var (
 		},
 		[]string{},
 	)
+	listStorageCount = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_list_storage_count",
+			Help:           "Number of LIST requests served from storage, split by resource and predicate complexity",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"resource", "predicate_complexity"},
+	)
+	listStorageQueries = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_list_storage_queries",
```

Comment:
this is very strange... this is a count of the paginated requests, right? why are we accumulating the pages per resource?

Reply:
This one I questioned previously too.
In theory, there is a constant cost associated with each etcd query. But having too much data points is making it harder to reason about things.

So I'm personally against introducing this metric.